src/falcor/transforms/index.js

```diff
@@ -196,3 +196,34 @@ export function remove ( collection, user, _id ) {
     });
 }
 
+export function archiveNode(nodeLabel, nodeId, userId) {
+  const query = `
+   MATCH (i:${nodeLabel} {_id:"${nodeId}"})
+   OPTIONAL MATCH (i)-[r]-(e)
+   SET i.archived = true, i.archived_at = ${Date.now()}, i.archiver = '${userId}'
+   return type(r) as type,i._id AS start,e._id AS end`
+  return this.flatMap(db =>
+    db.neo.run(query)
+      .flatMap(record =>
+        record.get('type') == null
+          ? Observable.of({start: record.get('start')})
+          : this::archiveRelationship(record.get('type'),record.get('start'),record.get('end'), userId)
```

Comment:
Clever, but you can actually do it all inside the cypher query above to avoid `n` trips to the db. e.g.:

```
MATCH (node:${nodeLabel} {_id:{nodeId}})
OPTIONAL MATCH (node)<-[rel]-()
SET node.archived = true,
  node.archived_at = timestamp(),
  node.archiver = {userId},
  rel.archived = true,
  rel.archived_at = timestamp(),
  rel.archiver = {userId}
RETURN node
```

Also notice that I am not using interpolation for most of the variables. That's because we would be subject to injection attacks if we dynamically compose our queries (though since this is internal, it is unlikely). The second argument to the `db.neo.run` function is a map of properties you want to pass, which you can then reference in the query with curly braces, as I've done above.